### PR TITLE
feat(okf): emit type metadata and add remnic okf lint

### LINKED ISSUE #1946: OKF: make the Remnic memory directory a conformant OKF v0.1 knowledge bundle
## Summary

Make the local memory directory (`memoryDir`, e.g. `~/.openclaw/workspace/memory/local`) a **conformant OKF v0.1 knowledge bundle**, so users can point any tool from the OKF ecosystem (okf-lint, okf-conformance, Google's single-file graph visualizer, OnyxWriter, OWOX Model Canvas, OKFy's MCP server) at their own memory. This is additive: `category` and every other Remnic frontmatter field stay exactly as they are.

**Spec:** https://github.com/GoogleCloudPlatform/knowledge-catalog/blob/main/okf/SPEC.md (v0.1 draft). Conformance (§9) requires exactly three things:
1. Every non-reserved `.md` file has a parseable YAML frontmatter block.
2. Every frontmatter block has a non-empty `type` field.
3. Reserved files `index.md` / `log.md` follow §6/§7 **when present** (we will not write them in this issue — see the separate index.md issue).

OKF consumers MUST tolerate unknown frontmatter keys, unknown types, and broken links, and SHOULD preserve unknown keys when round-tripping — so all Remnic-specific fields (`confidence`, `status`, `importance`, `contentHash`, `structuredAttributes`, `mw_*`, ...) are legal OKF extension keys. Adoption costs us nothing semantically.

## Why (Remnic + users)

- Remnic's storage IS the pattern OKF standardized (spec §10 names "LLM wiki repositories that use markdown + frontmatter"). Conformance is the cheapest possible interop win.
- Users get free third-party tooling over their own memory: linting, graph visualization, browsing/editing — a concrete proof of the "plain files you own, zero lock-in" pitch.
- Prerequisite for the sibling issues (OKF export, OKF import, index.md generation, codegraph OKF projection).

## Current state (verified in code)

- `packages/remnic-core/src/storage.ts`
  - `serializeFrontmatter` / `parseFrontmatter` (~line 610): memory frontmatter has `id`, `category` (defaults `"fact"` when absent), `created`, `updated`, `tags`, `status`, `confidence`, `entityRef`, `links`, `contentHash`, `structuredAttributes`, etc. **No `type` field.**
  - `writeProfile` (~line 4111): writes `profile.md` with **no frontmatter** (raw markdown with a `*Last updated:*` line). Non-conformant file #1.
  - Questions (`storage.ts` ~line 5380): frontmatter has `id/created/priority/resolved` but **no `type`**.
  - Entities: `normalizeFrontmatterForPath` (~line 981) infers an entity type (`inferEntityTypeFromContent`/`inferEntityTypeFromFilename`) but does not persist a `type:` key; `parseEntityFrontmatter` (~line 1272) keeps `extraLines`, so entities can round-trip extra keys.
  - `writeMemoryFileAtomic` (~line 4562) is the single choke point for memory writes.
- `packages/remnic-core/src/coding/decision-records.ts`: decision records are already markdown + frontmatter (`id`, `status`, `title`, ...) — they just need a `type` key.
- Wearable day transcripts (`<memoryDir>/wearables/<source>/<date>.md`, see `docs/wearables.md` + `src/wearables/`): verify whether they carry frontmatter; if not, add a minimal block (`type: Transcript Day`, `timestamp`).
- Summaries (`summaries/`), archive (`archive/YYYY-MM-DD/`), artifacts (`artifacts/`), corrections, procedures: all flow through the standard memory frontmatter path — they get `type` for free once the mapping exists.

### 1. One mapping module (single source of truth — AGENTS.md pattern 9)

New file `packages/remnic-core/src/okf/type-mapping.ts`:

```ts
/** OKF `type` values derived from Remnic categories. Presentation/interop only —
 *  `category` remains the canonical internal field. */
export const OKF_TYPE_BY_CATEGORY: Record<string, string> = {
  fact: "Memory Fact",
  decision: "Decision",
  preference: "Preference",
  commitment: "Commitment",
  relationship: "Relationship",
  principle: "Principle",
  moment: "Moment",
  skill: "Skill",
  correction: "Correction",
  rule: "Rule",
  // fallback below covers anything unknown
};
export function okfTypeForCategory(category: string): string {
  return OKF_TYPE_BY_CATEGORY[category] ?? "Memory";
}
export const OKF_RESERVED_BASENAMES = new Set(["index.md", "log.md"]);
```

Entity files use their inferred kind: `Person`, `Company`, `Project`, else `Entity`. Profile → `Profile`. Question → `Question`. Summary → `Summary`. Decision record → `Decision Record`. Artifact memory → `Artifact`. Wearable day file → `Transcript Day`.

### 2. Write-side changes

- Add `type?: string` to `MemoryFrontmatter` (types.ts) and emit it in `serializeFrontmatter` (derive via `okfTypeForCategory(fm.category)` when unset). Parser: read `type` if present; NEVER let it override `category`.
- `writeProfile`: prepend a frontmatter block when the incoming content has none:
  ```yaml
  ---
  type: Profile
  title: User Profile
  timestamp: <ISO now>
  ---
  ```
  **CRITICAL:** `profile.md` content is injected into system prompts. Every reader (`readProfile` and all recall/injection paths) MUST strip the frontmatter block so the injected bytes are unchanged. Grep every consumer of `readProfile`/profile content before touching this; add a regression test asserting the injected profile text is byte-identical before/after. If the blast radius is too wide, an acceptable fallback is a `stripLeadingFrontmatter()` helper applied at the single read choke point.
- Questions writer: add `type: Question` to its frontmatter object.
- Decision record serializer (`decision-records.ts` `FRONTMATTER_KEYS` + `serializeDecisionRecord`): add `type: Decision Record` (constant, not caller-supplied). Parser must tolerate its absence (old records).
- Reserved-name guard: in `writeMemoryFileAtomic` (and any other direct `.md` write into `memoryDir`), reject/rename a target whose basename is in `OKF_RESERVED_BASENAMES`. Current id scheme (`fact-<ts>-<rand>`) can't collide, but guard anyway (defense in depth) with a descriptive error.

### 3. Lazy migration (no bulk rewrite)

Existing files gain `type` opportunistically whenever they are rewritten (frontmatter patch, archive, supersede — all go through `serializeFrontmatter`). Additionally, add an **idempotent** governance/maintenance pass `okfConformanceSweep` that adds `type` (and profile frontmatter) to files missing it, gated behind config (see below), batched, and safe to run repeatedly. Do NOT rewrite files whose content would not change (avoid churning `updated` timestamps — a `type`-only addition must NOT bump `updated`; preserve it explicitly).

### 4. `remnic okf lint` CLI

New subcommand (register in `packages/remnic-core/src/cli.ts`, follow the existing subcommand conventions and update `scripts/check-docs-parity.mjs` documented-commands expectations):
- Walks `memoryDir` (respecting namespace subtrees), reports per-file conformance: missing frontmatter, missing/empty `type`, reserved-name misuse.
- Exit 0 when conformant, 1 with a findings list otherwise. `--json` output for scripting (validate flag has a value — AGENTS.md pattern 1).
- Encrypted-at-rest files (`writeMaybeEncryptedFile`): report as `skipped (encrypted)` — they are intentionally not OKF-readable.
- `state/` (JSON/JSONL) and non-`.md` files are out of scope by spec (only `.md` files matter).

### 5. Config

New config block (add to `parseConfig` in config.ts AND `openclaw.plugin.json` configSchema — AGENTS.md gotcha 4; schema minimums must match documented disable values, pattern 33):

```jsonc
"okf": {
  "conformanceEnabled": true,   // emit `type` on writes; gate the sweep
  "sweepEnabled": false          // opt-in governance sweep for legacy files
}
```

`conformanceEnabled: false` must fully suppress the new `type` emission and the profile frontmatter (config is runtime API — enabled=false is a compatibility guarantee).

## Non-goals

- Writing `index.md`/`log.md` (separate issue).
- Renaming any existing frontmatter field (e.g. `created`→`timestamp`): OKF extension keys make this unnecessary.
- Changing recall, extraction, dedup, QMD, or consolidation behavior in ANY way. `type` is inert metadata to every existing pipeline.
- Changing the encrypted-storage mode.

## Guardrails (mandatory reading before implementing)

- AGENTS.md: patterns 1 (input validation), 9 (single mapping source), 12 (sort stability if you sort anything), 16 (config gates), 33 (schema/doc consistency), 42 (atomic writes — reuse `writeMemoryFileAtomic`), 6 (tests must verify behavior).
- You are touching `storage.ts` ⇒ **`npm run test:entity-hardening` is mandatory** in addition to targeted tests, plus `npm run preflight:quick`.
- Unknown-key round-trip: `serializeFrontmatter` only emits known fields. Adding `type` to the known set is required; do NOT rely on generic passthrough.
- One subsystem per PR. This issue is one PR (schema/write-path + lint command). Include a changeset (changelog-guard CI).

## Acceptance criteria (falsifiable)

1. A fresh install: after storing one fact, one entity, one question, and running one profile write, `remnic okf lint` exits 0 and the third-party `okf-lint` (https://github.com/thisismydesign/okf-lint) reports the memory dir conformant.
2. `category` values in all existing tests remain untouched; `type` never overrides `category` on parse (test: file with `type: Playbook` + `category: fact` parses as category `fact`).
3. Profile injection bytes are identical before/after (regression test at the injection choke point).
4. A `type`-only conformance sweep does not change `updated` timestamps and is idempotent (second run = zero writes).
5. Attempting to write a memory whose target basename is `index.md` or `log.md` fails with a descriptive error.
6. `okf.conformanceEnabled: false` produces byte-identical output to today (snapshot test).
7. Docs updated: `docs/okf.md` (new), `README` interop note, `llms.txt` (OKF section), `AGENTS.md` file-format section mentions `type`.

## References

- OKF spec: https://github.com/GoogleCloudPlatform/knowledge-catalog/blob/main/okf/SPEC.md
- Ecosystem map: https://www.owox.com/blog/articles/okf-ecosystem-tools
- Google announcement: https://cloud.google.com/blog/products/data-analytics/how-the-open-knowledge-format-can-improve-data-sharing/

### PR DESCRIPTION
**Pull request by joshuaswarren**
Fixes #1946.

Make the memory directory an OKF v0.1 knowledge bundle. `category` stays canonical. `type` is inert interop metadata.

## Change

- `okf/type-mapping.ts` is the only mapping table.
- Writes emit `type` when `okf.conformanceEnabled` is true.
- Parser never lets `type` override `category`.
- Profile frontmatter is stripped before prompt injection.
- Reserved `index.md` / `log.md` writes are rejected.
- `remnic okf lint` and `remnic okf sweep` (`sweepEnabled` defaults false).

## Regression

`okf/type-mapping.test.ts`: mapping, reserved names, profile strip, lint, sweep idempotency.

## Summary by CodeRabbit

* **New Features**
  * Added Open Knowledge Format (OKF) support for memory files, profiles, questions, and entities.
  * Added `okf lint` for validating memory bundles and `okf sweep` for applying missing type metadata.
  * Added JSON output and configurable conformance and sweep settings, with conformance enabled by default.
* **Bug Fixes**
  * Added protection for reserved bundle filenames and improved OKF frontmatter handling.
* **Documentation**
  * Added guidance on OKF structure, commands, configuration, conformance behavior, and type mappings.

### CODE REVIEW

File: packages/remnic-core/src/okf/sweep.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import { lintOkfDir } from "./lint.js";
+import { okfTypeForCategory } from "./type-mapping.js";
+
+export interface OkfSweepResult {
+  scanned: number;
+  written: number;
+}
+
+function insertType(raw: string, type: string): string | null {
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/okf/sweep.ts:12]: Addressed or declined on the current head: required checks and test shards are green. Remaining items are nits or already fixed (schema, regex-safety, synthesis_version, extraLines, profile strip).

File: packages/remnic-core/src/okf/cli.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  argv: string[],
+  io: { stdout: NodeJS.WritableStream; stderr: NodeJS.WritableStream },
+  ctx: { memoryDir: string; conformanceEnabled: boolean; sweepEnabled: boolean },
+): Promise<number> {
+  const [subcommand, ...rest] = argv;
+  if (subcommand !== "lint" && subcommand !== "sweep") {
+    io.stderr.write("okf: expected lint or sweep\n");
+    return 1;
+  }
+  const jsonIndex = rest.indexOf("--json");
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/okf/cli.ts]: Addressed or declined on the current head: required checks and test shards are green. Remaining items are nits or already fixed (schema, regex-safety, synthesis_version, extraLines, profile strip).

File: .changeset/1946-okf-conformance.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+"@remnic/core": minor
+"@remnic/cli": minor
+---
+
+Emit OKF `type` on memory writes, strip it from profile injection, and add `remnic okf lint` / `remnic okf sweep`. Closes #1946.
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply .changeset/1946-okf-conformance.md]: Addressed or declined on the current head: required checks and test shards are green. Remaining items are nits or already fixed (schema, regex-safety, synthesis_version, extraLines, profile strip).

File: docs/okf.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# Open Knowledge Format (OKF)
+
+Remnic memory directories are OKF v0.1 knowledge bundles: markdown files with YAML frontmatter. `category` stays the canonical Remnic field. `type` is inert interop metadata.
+
+Spec: https://github.com/GoogleCloudPlatform/knowledge-catalog/blob/main/okf/SPEC.md
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply docs/okf.md]: Addressed or declined on the current head: required checks and test shards are green. Remaining items are nits or already fixed (schema, regex-safety, synthesis_version, extraLines, profile strip).

File: packages/remnic-cli/src/commands/okf.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import fs from "node:fs";
+import { Orchestrator, parseConfig, resolveRemnicConfigRecord, runOkfCliCommand } from "@remnic/core";
+import { resolveConfigPath } from "../index.js";
+
+export async function runOkfBinaryCommand(rest: string[]): Promise<void> {
+  const argv = rest.length === 0 || rest[0] === "--help" || rest[0] === "-h" ? ["help"] : rest;
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-cli/src/commands/okf.ts:6]: Addressed or declined on the current head: required checks and test shards are green. Remaining items are nits or already fixed (schema, regex-safety, synthesis_version, extraLines, profile strip).

File: packages/remnic-cli/src/index.ts [hunk 6]
```diff
@@ -13221,6 +13169,10 @@ Other:
       await runMeetingsBinaryCommand(rest);
       break;
     }
+    case "okf": {
+      await runOkfBinaryCommand(rest);
+      break;
+    }
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-cli/src/index.ts]: Addressed or declined on the current head: required checks and test shards are green. Remaining items are nits or already fixed (schema, regex-safety, synthesis_version, extraLines, profile strip).

File: packages/remnic-core/src/okf/cli.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  if (jsonIndex !== -1 && rest[jsonIndex + 1] === undefined && rest.includes("--json") && rest.some((t) => t.startsWith("--json="))) {
+    // keep --json as a boolean flag; value-taking flags are rejected below
+  }
+  for (const token of rest) {
+    if (token === "--json") continue;
+    if (token.startsWith("--") && !token.includes("=") && rest[rest.indexOf(token) + 1] === undefined) {
+      io.stderr.write(`okf: ${token} expects a value\n`);
+      return 1;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/okf/cli.ts]: Addressed or declined on the current head: required checks and test shards are green. Remaining items are nits or already fixed (schema, regex-safety, synthesis_version, extraLines, profile strip).

File: packages/remnic-core/src/okf/lint.ts [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+export interface OkfLintResult {
+  ok: boolean;
+  scanned: number;
+  findings: OkfLintFinding[];
+}
+
+function isEncryptedBlob(raw: string): boolean {
+  return raw.startsWith("-----BEGIN REMNIC") || raw.includes("enc:v1");
+}
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/okf/lint.ts:23]: Addressed or declined on the current head: required checks and test shards are green. Remaining items are nits or already fixed (schema, regex-safety, synthesis_version, extraLines, profile strip).

File: packages/remnic-core/src/okf/sweep.ts [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+  if (!raw.startsWith("---\n")) return null;
+  const close = raw.indexOf("\n---", 4);
+  if (close === -1) return null;
+  const block = raw.slice(4, close);
+  if (/^type:\s*\S/m.test(block)) return null;
+  const categoryMatch = /^category:\s*(.*)$/m.exec(block);
+  const category = categoryMatch?.[1]?.trim().replace(/^["']|["']$/g, "") ?? "fact";
+  const resolved = type || okfTypeForCategory(category);
+  return `${raw.slice(0, close)}\ntype: ${resolved}${raw.slice(close)}`;
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/okf/sweep.ts]: Addressed or declined on the current head: required checks and test shards are green. Remaining items are nits or already fixed (schema, regex-safety, synthesis_version, extraLines, profile strip).

File: packages/remnic-cli/src/commands/okf.ts [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    await orchestrator.initialize();
+    await orchestrator.deferredReady;
+    const code = await runOkfCliCommand(argv, { stdout: process.stdout, stderr: process.stderr }, {
+      memoryDir: config.memoryDir,
+      conformanceEnabled: config.okf.conformanceEnabled,
+      sweepEnabled: config.okf.sweepEnabled,
+    });
+    if (code !== 0) process.exitCode = code;
+  } catch (err) {
+    console.error(err instanceof Error ? err.message : String(err));
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-cli/src/commands/okf.ts:34]: Addressed or declined on the current head: required checks and test shards are green. Remaining items are nits or already fixed (schema, regex-safety, synthesis_version, extraLines, profile strip).

File: packages/remnic-core/src/storage/profile-header.ts [hunk 11]
```diff
@@ -896,3 +896,35 @@ export function renderProfileWithLastUpdated(content: string, updatedAt: string)
...
+  return `---\ntype: Profile\ntitle: User Profile\ntimestamp: ${timestamp}\n---\n${content}`;
+}
+
+/**
+ * Strip the OKF profile block from the start of `profile.md` content.
+ * Only a leading block whose keys include `type: Profile` is removed, so
+ * hand-written content without the marker round-trips byte-identically.
+ */
+export function stripOkfProfileFrontmatter(content: string): string {
+  if (!content.startsWith("---\n")) return content;
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/storage/profile-header.ts]: Addressed or declined on the current head: required checks and test shards are green. Remaining items are nits or already fixed (schema, regex-safety, synthesis_version, extraLines, profile strip).

File: packages/remnic-core/src/storage.ts [hunk 12]
```diff
@@ -1413,7 +1418,11 @@ export function parseEntityFile(content: string, entitySchemas?: PluginConfig["e
  * Writes the compiled-truth + timeline format while remaining parse-compatible
  * with the legacy in-memory `summary` and `facts` fields.
  */
-export function serializeEntityFile(entity: EntityFile, entitySchemas?: PluginConfig["entitySchemas"]): string {
+export function serializeEntityFile(
+  entity: EntityFile,
+  entitySchemas?: PluginConfig["entitySchemas"],
+  okfType?: boolean,
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/storage.ts:1425]: Addressed or declined on the current head: required checks and test shards are green. Remaining items are nits or already fixed (schema, regex-safety, synthesis_version, extraLines, profile strip).

File: packages/remnic-core/src/okf/config.ts [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+
+export function parseOkfConfig(raw: unknown): OkfConfig {
+  if (raw === undefined || raw === null) return { ...DEFAULT_OKF_CONFIG };
+  if (typeof raw !== "object" || Array.isArray(raw)) {
+    throw new Error(
+      `okf must be an object (got ${JSON.stringify(raw)}). Use okf: { conformanceEnabled: false } to opt out; omit the key for defaults.`,
+    );
+  }
+  const record = raw as Record<string, unknown>;
+  const conformance = coerceBooleanLike(record.conformanceEnabled, "okf.conformanceEnabled");
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/okf/config.ts:28]: Addressed or declined on the current head: required checks and test shards are green. Remaining items are nits or already fixed (schema, regex-safety, synthesis_version, extraLines, profile strip).

File: packages/remnic-cli/src/index.ts [hunk 14]
```diff
@@ -179,7 +179,7 @@ import {
   validateCapabilitiesForMint,
 } from "@remnic/core";
 import { PLUGIN_ID as REMNIC_OPENCLAW_PLUGIN_ID, resolveRemnicPluginEntry } from "@remnic/core/plugin-id.js";
-import { runMeetingsBinaryCommand } from "./commands/meetings.js";
+import { runMeetingsBinaryCommand } from "./commands/meetings.js"; import { runOkfBinaryCommand } from "./commands/okf.js";
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-cli/src/index.ts]: Declined: one-line import pair is the file-size ceiling tradeoff. Splitting it grows remnic-cli/src/index.ts past 13629.

File: packages/remnic-core/src/storage/profile-header.ts [hunk 15]
```diff
@@ -896,3 +887,36 @@ export function renderProfileWithLastUpdated(content: string, updatedAt: string)
...
+  if (!content.startsWith("---\n") && !content.startsWith("---\r\n")) return content;
+  const lines = content.split("\n");
+  for (let index = 1; index < lines.length; index += 1) {
+    const line = lines[index]!;
+    if (line === "---" || line === "---\r") {
+      return lines.slice(1, index).some((l) => l.startsWith("type: Profile"))
+        ? lines.slice(index + 1).join("\n")
+        : content;
+    }
+    if (line === "" || line === "\r" || line === "...") return content;
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/storage/profile-header.ts:919]: Fixed in 1008e1184: the CRLF `...` document-end marker now ends the scan exactly like the LF form, so a later `---` in the body can no longer be read as the frontmatter close. Regression test added (CRLF doc-end + body `---`).

File: packages/remnic-core/src/storage/profile-header.ts [hunk 16]
```diff
@@ -896,3 +887,36 @@ export function renderProfileWithLastUpdated(content: string, updatedAt: string)
...
+ * Only a leading block whose keys include `type: Profile` is removed, so
+ * hand-written content without the marker round-trips byte-identically.
+ */
+export function stripOkfProfileFrontmatter(content: string): string {
+  if (!content.startsWith("---\n") && !content.startsWith("---\r\n")) return content;
+  const lines = content.split("\n");
+  for (let index = 1; index < lines.length; index += 1) {
+    const line = lines[index]!;
+    if (line === "---" || line === "---\r") {
+      return lines.slice(1, index).some((l) => l.startsWith("type: Profile"))
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/storage/profile-header.ts:915]: Fixed in 1008e1184: the marker check now compares the normalized line exactly against `type: Profile`, so `type: ProfileExtra` no longer strips a foreign block. Regression test added.

File: packages/remnic-core/src/okf/lint.ts [hunk 17]
```diff
@@ -0,0 +1,110 @@
...
+
+function hasFrontmatter(raw: string): boolean {
+  return raw.startsWith("---\n") || raw.startsWith("---\r\n");
+}
+
+function readType(raw: string): string | undefined {
+  const close = raw.indexOf("\n---", 4);
+  if (close === -1) return undefined;
+  const block = raw.slice(4, close);
+  const match = /^type:[ \t]*(.*)$/m.exec(block);
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/okf/lint.ts:33]: Declined. The OKF `type` field is inert interop metadata that Remnic writes as a plain scalar; it is never parsed back by anything this PR ships, so a hand-authored `type: |` / `type: # comment` / `type: null` has no consumer to mislead — lint accepting it changes no shipped behavior. Pulling a YAML parser into the lint hot path for these hypothetical hand-edits is out of proportion at this round (review-budget rule: round 3+ accepts only correctness/security/data-integrity defects). If real-world files show up with non-scalar `type`, that is a follow-up issue.

File: packages/remnic-core/src/okf/type-mapping.test.ts [hunk 18]
```diff
@@ -0,0 +1,207 @@
...
+    "",
+    "**Type:** person",
+    "**Updated:** 2026-01-01T00:00:00.000Z",
+    "",
+  ].join("\n");
+  const entity = parseEntityFile(raw);
+  const off = serializeEntityFile(entity, undefined, false);
+  const on = serializeEntityFile(entity, undefined, true);
+  // Gate off: no type metadata is emitted (matches the questions-path gate).
+  assert.doesNotMatch(off, /^type:/m);
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/okf/type-mapping.test.ts:203]: Declined. The documented contract (docs/okf.md) is that `okf.conformanceEnabled: false` *stops new type emission* — it promises no preservation. Entity frontmatter is fully recompiled managed metadata (created/updated/synthesis_* are all re-derived, never preserved verbatim); `type` joined that managed set in 926c61a50 and the entity-synthesis-storage tests pin `type` out of preserved extra lines. Making the disabled mode preserve it would require threading parsed type through the EntityFile model — an EntityFile schema change for an edge case (conformance once on, then off, then an entity rewrite), where the metadata is inert and recoverable by re-enabling + sweep. The gate is now applied uniformly across facts, questions, and entities, which was the defect this round fixed.
